### PR TITLE
Adding Preshower detector (PSR) for ALICE3 simulation

### PR DESCRIPTION
**Pull request by an15ms157**
This PR creates a simple 8 layered preshower detector named PSR. This is a cylindrical detector of length 100 cm whose shower layers are made up of Pb (0.5 cm) and the detector layers are made up of Si (45 microns). PSR is based on the ITSMFT classes and the code is structurally similar to that of FT3. Each layer is made of a monolithic silicon disk with a thin sensitive layer for hit generation. Silicon chip thickness is tuned to match the layer x/X0 to allow a minimal evaluation of material budget effects.

One should get a file o2sim_HitsPSR.root by running
$ o2-sim -m PSR -e TGeant3 -g boxgen -n 10

Another PR of the same name exists here which might be deleted later: https://github.com/AliceO2Group/AliceO2/pull/6519

### CODE REVIEW

**Comment by mconcas** [comment]
Thanks. Please consider to apply this patch: [patch.txt](https://github.com/AliceO2Group/AliceO2/files/6867950/patch.txt)
To fix clang format test.

Cheers

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID.h [hunk 1]
```diff
@@ -24,7 +24,8 @@
   }
   printf("joint mask: 0x%lx\n",mskTot.to_ulong());
  */
-
+//
```
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID.h:27]: Do we really need these lines?

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_BACKUP_450708.h [hunk 2]
```diff
@@ -0,0 +1,226 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
```
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_BACKUP_450708.h]: I guess this file is not needed

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_BASE_450708.h [hunk 3]
```diff
@@ -0,0 +1,173 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
```
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_BASE_450708.h]: same ?

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_LOCAL_450708.h [hunk 4]
```diff
@@ -0,0 +1,211 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
```
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_LOCAL_450708.h]: same

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_REMOTE_450708.h [hunk 5]
```diff
@@ -0,0 +1,174 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
```
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID_REMOTE_450708.h]: same

File: Detectors/Upgrades/ALICE3/FT3/simulation/src/Detector.cxx [hunk 6]
```diff
@@ -160,7 +160,7 @@ void Detector::buildBasicFT3(const FT3BaseParam& param)
   mLayerID.clear();
   mLayers.resize(2);
 
-  for (Int_t direction : {0, 1}) {
+  for (Int_t direction : {1}) {
```
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/FT3/simulation/src/Detector.cxx]: same comment for old PR

File: Detectors/Upgrades/ALICE3/PSR/README.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<!-- doxy
```
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/PSR/README.md:1]: Update for PSR please

**Review by mconcas — CHANGES_REQUESTED** [review]
Hi, inline you find some comments. Overall seems ok.

File: Detectors/Upgrades/ALICE3/PSR/README.md (hunk 7 above)
Comment (an15ms157) [reply Detectors/Upgrades/ALICE3/PSR/README.md:1]: > Update for PSR please

@mconcas , I have applied the patch, deleted the unnecessery file made all the required changes. However, I am not sure what to do about these lines. Is it okay if I just keep the description of the Pre-Shower detector here? 

Once done, I plan to make a commit, merge it with the initial commit and push it again to my repo.

File: Detectors/Upgrades/ALICE3/PSR/README.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+\page refDetectorsUpgradesALICE3FT3 EndCaps
+/doxy -->
+
+# PostLS4EndCaps
```
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/PSR/README.md]: Endcaps -> Preshower

File: Detectors/Upgrades/ALICE3/PSR/README.md (hunk 7 above)
Comment (mconcas) [reply Detectors/Upgrades/ALICE3/PSR/README.md:1]: You don't need to merge any commit, just put the fixing commit on top.
For those lines just make sure to change any reference to FT3 to PSR

File: Detectors/Upgrades/ALICE3/PSR/README.md (hunk 8 above)
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/PSR/README.md]:
```suggestion
# PostLS4Preshower
```

File: Detectors/Upgrades/ALICE3/PSR/README.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<!-- doxy
+\page refDetectorsUpgradesALICE3FT3 EndCaps
```
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/PSR/README.md]:
```suggestion
\page refDetectorsUpgradesALICE3PSR Preshower
```

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID.h (hunk 1 above)
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID.h:27]:
```suggestion

```

File: Detectors/Upgrades/ALICE3/FT3/simulation/src/Detector.cxx (hunk 6 above)
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/FT3/simulation/src/Detector.cxx]:
```suggestion
  for (Int_t direction : {0, 1}) {
```

File: macro/build_geometry.C [hunk 10]
```diff
@@ -202,6 +203,16 @@ void build_geometry(FairRunSim* run = nullptr)
     auto ft3 = new o2::ft3::Detector(true);
     run->AddModule(ft3);
   }
+  if (isActivated("PSR")) {
```
Comment (mconcas) [inline macro/build_geometry.C]: Possibly a duplication?

File: DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID.h [hunk 11]
```diff
@@ -12,19 +12,7 @@
 /// @brief ALICE detectors ID's, names, masks
 ///
 /// @author Ruben Shahoyan, ruben.shahoyan@cern.ch
-
```
Comment (mconcas) [inline DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/DetID.h]: Why this deletion?

**Review by mconcas — DISMISSED** [review]

**Review by marcovanleeuwen — DISMISSED** [review]

**Review by mconcas — APPROVED** [review]
Lgtm

**Comment by mconcas** [comment]
@sawenzel : if it is ok with you we are ready to merge

**Comment by ktf** [comment]
Any reason why this is not yet merged?

**Comment by mconcas** [comment]
> Any reason why this is not yet merged?

As it changes the `build_geometry.C` macro I was interested in @sawenzel feedback.
In the meantime the inclusion of the PSR detector in O2 ALICE 3 simulation seemed to be not critical/a blocker (at least until now), so I am actually closing it for the time being to avoid unused code addition.
If it becomes relevant again I'll reopen.

**Pull request closed without merging**